app/modules/intelligence/agents/chat_agents/pydantic_agent.py

```
@@ -159,93 +202,354 @@ async def run(self, ctx: ChatContext) -> ChatAgentResponse:
-                                    event.part, TextPart
-                                ):
+            # Try to initialize MCP servers with timeout handling
+            try:
+                async with agent.run_mcp_servers():
+                    async with agent.iter(
+                        user_prompt=ctx.query,
+                        message_history=[
+                            ModelResponse([TextPart(content=msg)])
+                            for msg in ctx.history
+                        ],
+                    ) as run:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)